docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+            export_model.save(f"{model_save_loc}-exported", save_format="tf", overwrite=True)
+        except Exception as e:
+            logger.error(f"ERR in TRAIING \n {e}")
+            raise e
+```
+
+While this method may be intimidating, don't let it scare you! It is simply the core code of our companion piece, gently refactored to fit into our tool's (and the PythonSdk's) runtime and lifecycle.
+
+In this method we:
+  * load our dataset `tf.data.Dataset.save's sibling
+  * create and adapt our `vectorize_layer: tensorflow.keras.layers.TextVectorization` as recommended by TensorFlow
+  * cache our datasets (now loading data as `Datasets` begins to pay off!)
+  * init, compile, evaluate, and save our base model.]
+  * generate  data for our previews
+
+You may have noticed we use a decorator here as well `static method`.
+We do this to allow us to call this using `Multiprocessing`  to train our model without blocking our Python sdk service process. We also add in a multiprocessing queue to quickly retrieve the results and send them to the front end before ending.
+Lastly, note that we catch a generalized exception and explicitly raise it for this method. Doing that allows us to throw in a controlled way such that the Designer can report but log it _before_ the process terminates, as we discussed early on in this section of the guide.
+
+As you might have noticed, we currently have a non-trivial amount of arguments.
+Since we collect these via IPC (**Read:** InterProcess Communication: N processes running separately; think to Excel and tablue with a linked spreadsheet.), these will also be more difficult to keep in sync.
+As such, we want a function to extract these that we can wrap in a try/catch block to explicitly capture and report `KeyErrors` or similarly common exceptions in this context.
+
+
+#### Function: `get_model_args()`
+
+```python
+def get_model_args(self):
+  try:
+    root_url = self.data_url
+    max_features = int(self.provider.tool_config["trainingConfig"]["maxFeatures"])
+    seq_length = int(self.provider.tool_config["textVectorizationConfig"]["sequenceLength"])
+    # Make a text-only dataset (without labels), then call adapt
+    embedding_dim = int(self.provider.tool_config["modelConfig"]["embeddingDim"])
+    model_save_loc = self.provider.tool_config["modelConfig"]["modelName"]
+    epochs = int(self.provider.tool_config["trainingConfig"]["epochs"])
+  except Exception as e:
+    self.info(str(e)) # Editing(E)-NOTE: warn method?
+    self.provider.io.error(f"An invalid key or value was provided: {str(e)}")
+  return (
+    root_url,
+    max_features,
+    seq_length,
+    embedding_dim,
+    model_save_loc,
+    epochs
+  )
+```
+
+The method above should be pretty straightforward now; good ol' dictionary access. (E-Note: Idiom safe?)
+Otherwise, carefully note what we do and do not cast.
+Consider why this may be relative to the subsequent code updates in `init` and `on_complete`.
```

Comment:
TW review:

The method above should be pretty straightforward now, standard dictionary access. Otherwise, carefully note what we do and do not cast. Consider why this might be relative to the subsequent code updates in `init` and `on_complete`.